### PR TITLE
Fix couple of nits in `nodevolumelimits` plugin

### PR DESCRIPTION
**Pull request by chendave**
Signed-off-by: Dave Chen <dave.chen@arm.com>

#### What type of PR is this?

/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/scheduler/framework/plugins/nodevolumelimits/csi.go [hunk 1]
```diff
@@ -75,7 +75,7 @@ func (pl *CSILimits) Filter(ctx context.Context, _ *framework.CycleState, pod *v
 
 	node := nodeInfo.Node()
 	if node == nil {
-		return framework.NewStatus(framework.Error, fmt.Sprintf("node not found"))
+		return framework.NewStatus(framework.Error, "node not found")
```
Comment (damemi) [inline pkg/scheduler/framework/plugins/nodevolumelimits/csi.go:78]: could these `Sprintfs` actually give some info from the `nodeInfo` about which node was not found?

Comment (chendave) [reply pkg/scheduler/framework/plugins/nodevolumelimits/csi.go:78]: that should be better.

Comment (chendave) [reply pkg/scheduler/framework/plugins/nodevolumelimits/csi.go]: Done

**Comment by chendave** [comment]
@damemi , thanks Mike for the review, updated.

File: pkg/scheduler/framework/plugins/nodevolumelimits/csi.go [hunk 2]
```diff
@@ -155,7 +155,7 @@ func (pl *CSILimits) filterAttachableVolumes(
 
 		driverName, volumeHandle := pl.getCSIDriverInfo(csiNode, pvc)
 		if driverName == "" || volumeHandle == "" {
-			klog.V(5).Infof("Could not find a CSI driver name or volume handle, not counting volume")
```
Comment (damemi) [inline pkg/scheduler/framework/plugins/nodevolumelimits/csi.go:158]: Actually, these could be changed to structured logging `InfoS` (and below, the `Errorf` could be `ErrorS`) (see https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1602-structured-logging)

Comment (chendave) [reply pkg/scheduler/framework/plugins/nodevolumelimits/csi.go:158]: Yes, but for those cases, I cannot see any benefit after migrating to structured logging, 

see the non-goals in that spec:
https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1602-structured-logging#non-goals

> We are not replacing currently used logging library (klog) or the way in which it is used
We are not preserving full compatibility of logging output
We are not proposing structuring all logs in Kubernetes

Comment (chendave) [reply pkg/scheduler/framework/plugins/nodevolumelimits/csi.go:158]: unless `klog` is going to be deprecated and retied, I think we won't bother to migrating those.

**Comment by chendave** [comment]
@damemi Does my reply make sense? or do you still think it's better to switch to structured log? thanks!

File: pkg/scheduler/framework/plugins/nodevolumelimits/non_csi.go [hunk 3]
```diff
@@ -344,7 +344,7 @@ func getMaxVolLimitFromEnv() int {
 		if parsedMaxVols, err := strconv.Atoi(rawMaxVols); err != nil {
 			klog.Errorf("Unable to parse maximum PD volumes value, using default: %v", err)
```
Comment (damemi) [inline pkg/scheduler/framework/plugins/nodevolumelimits/non_csi.go:345]: @chendave what do you think about adding this to the PR to be `ErrorS`?

I see what you mean about the others not benefitting from it. But while we are making a change here, might as well throw this one in too. Sorry for the back and forth

**Comment by chendave** [comment]
Found a chasing here, https://github.com/kubernetes/kubernetes/pull/100947
will rebase once that got merged.

File: pkg/scheduler/framework/plugins/nodevolumelimits/non_csi.go (hunk 3 above)
Comment (chendave) [reply pkg/scheduler/framework/plugins/nodevolumelimits/non_csi.go:345]: Done

**Comment by chendave** [comment]
rebased, thanks @damemi for the review.

File: pkg/scheduler/framework/plugins/nodevolumelimits/csi.go [hunk 4]
```diff
@@ -75,14 +75,14 @@ func (pl *CSILimits) Filter(ctx context.Context, _ *framework.CycleState, pod *v
 
 	node := nodeInfo.Node()
 	if node == nil {
-		return framework.NewStatus(framework.Error, "node not found")
+		return framework.NewStatus(framework.Error, fmt.Sprintf("node not found: %s", node.Name))
```
Comment (chendave) [inline pkg/scheduler/framework/plugins/nodevolumelimits/csi.go:78]: unnecessary "fmt.Sprintf" has been removed by https://github.com/kubernetes/kubernetes/pull/100947, so rebased.

Comment (damemi) [reply pkg/scheduler/framework/plugins/nodevolumelimits/csi.go:78]: I see, it was unnecessary before when it wasn't logging the `node.Name`. Now that we are adding that, `Sprintf` is useful again.

**Review by damemi — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**